pkg/testcase/secretsencrypt.go

```diff
@@ -0,0 +1,137 @@
+		shared.LogLevel("INFO", "Sleep 10 seconds - wait before restarting next node in cluster")
+		time.Sleep(10 * time.Second)
+	}
+	switch product {
+	case "k3s":
+		shared.LogLevel("INFO", "Sleep 30 seconds - wait for services to come up")
+		time.Sleep(30 * time.Second)
+	case "rke2":
+		shared.LogLevel("INFO", "Sleep 60 seconds - wait for services to come up")
+		time.Sleep(60 * time.Second)
+	}
+
+	stdStatusOut, errStatus := shared.SecretEncryptOps("status", cpIp, product)
+	Expect(errStatus).NotTo(HaveOccurred(), "error getting secret-encryption status")
+	verifyStatusOutput(action, stdStatusOut)
+
+	errLog := logEncryptionFileContents(ips, product)
+	Expect(errLog).NotTo(HaveOccurred())
+}
+
+func verifyStdOut(action, stdout string) {
+	switch action {
+	case "prepare":
+		Expect(stdout).To(ContainSubstring("prepare completed successfully"))
+	case "rotate":
+		Expect(stdout).To(ContainSubstring("rotate completed successfully"))
+	case "reencrypt":
+		Expect(stdout).To(ContainSubstring("reencryption started"))
+	case "rotate-keys":
+		Expect(stdout).To(ContainSubstring("keys rotated, reencryption started"))
+	}
+}
+
+func verifyStatusOutput(action, stdout string) {
+	switch action {
+	case "prepare":
+		Expect(stdout).To(ContainSubstring("Encryption Status: Enabled"))
+		Expect(stdout).To(ContainSubstring("Current Rotation Stage: prepare"))
+		Expect(stdout).To(ContainSubstring("Server Encryption Hashes: All hashes match"))
+	case "rotate":
+		Expect(stdout).To(ContainSubstring("Encryption Status: Enabled"))
+		Expect(stdout).To(ContainSubstring("Current Rotation Stage: rotate"))
+		Expect(stdout).To(ContainSubstring("Server Encryption Hashes: All hashes match"))
+	case "reencrypt":
+		Expect(stdout).To(ContainSubstring("Encryption Status: Enabled"))
+		Expect(stdout).To(ContainSubstring("Current Rotation Stage: reencrypt_finished"))
+		Expect(stdout).To(ContainSubstring("Server Encryption Hashes: All hashes match"))
+	case "rotate-keys":
+		Expect(stdout).To(ContainSubstring("Encryption Status: Enabled"))
+		Expect(stdout).To(ContainSubstring("Current Rotation Stage: reencrypt_finished"))
+		Expect(stdout).To(ContainSubstring("Server Encryption Hashes: All hashes match"))
+	}
+}
+
+func getNodeIps(etcdNodes, cpNodes []shared.Node) []string {
+	var sarray []string
+	for _, node := range etcdNodes {
+		sarray = append(sarray, node.ExternalIP)
+	}
+	for _, node := range cpNodes {
+		sarray = append(sarray, node.ExternalIP)
+	}
+	return sarray
+}
+
+func logEncryptionFileContents(ips []string, product string) error {
+	configFile := fmt.Sprintf("/var/lib/rancher/%s/server/cred/encryption-config.json", product)
+	stateFile := fmt.Sprintf("/var/lib/rancher/%s/server/cred/encryption-state.json", product)
+	cmdShowConfig := fmt.Sprintf("sudo cat %s", configFile)
+	cmdShowState := fmt.Sprintf("sudo cat %s", stateFile)
+	for _, ip := range ips {
+		sout, errConfig := shared.RunCommandOnNode(cmdShowConfig, ip)
```

Comment:
what would be sout ?  can we name it better ?

Reply:
will rename it to configStdOut